swift-extensions/UISwitchExtensions.swift

```diff
@@ -15,21 +15,39 @@ extension UISwitch {
             viewModel = value
             guard let toggleSwitchViewModel = value else { return }
 
-            self.addTarget(self, action: #selector(UISwitch.onValueChanged), for: .valueChanged)
-            observe(toggleSwitchViewModel.isOn) { [weak self] (value: Bool) in
+            observe(PublisherExtensionsKt.distinctUntilChanged(toggleSwitchViewModel.isOn)) { [weak self] (value: Bool) in
                 self?.isOn = value
             }
 
-            bind(toggleSwitchViewModel.enabled, \UISwitch.isEnabled)
+            bind(toggleSwitchViewModel.isEnabled, \UISwitch.isEnabled)
+            
+            observe(toggleSwitchViewModel.toggleSwitchAction) {[weak self] (value: ViewModelAction) in
+                guard let self = self else { return }
+                self.removePreviousRegisteredTapAction()
+                if value != ViewModelAction.Companion().None {
+                    self.addTarget(self, action: #selector(UISwitch.onValueChanged), for: .valueChanged)
+                }
+            }
         }
     }
-
+    
+    private func removePreviousRegisteredTapAction() {
+        self.removeTarget(self, action: nil, for: .valueChanged)
+    }
+    
     @objc
     func onValueChanged(sender: UISwitch) {
         if #available(iOS 10.0, *), let style = impactFeedbackStyle() {
             UIImpactFeedbackGenerator(style: getFeedbackStyle(style: style)).impactOccurred()
         }
-        toggleSwitchViewModel?.setIsOn(on: self.isOn)
```

Comment:
I might be missing something but, since you are still not calling `toggleSwitchViewModel?.setIsOn(on: self.isOn)` how is the common code informed when the UI changes state ?